In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [6]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv.csv')

# Q4

In [8]:
#looking at top 5 rows of the dataframe(df)
gdp_df.head()

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2021  1517.016266              NaN
1     Afghanistan  2020  1968.341002              NaN
2     Afghanistan  2019  2079.921861              NaN
3     Afghanistan  2018  2060.698973              NaN
4     Afghanistan  2017  2096.093111              NaN

In [9]:
#looking at bottom 5 rows of the df
gdp_df.tail()

Country or Area  Year        Value  Value Footnotes
7657        Zimbabwe  1994  2670.106615              NaN
7658        Zimbabwe  1993  2458.783255              NaN
7659        Zimbabwe  1992  2468.278257              NaN
7660        Zimbabwe  1991  2781.787843              NaN
7661        Zimbabwe  1990  2704.757299              NaN

# Q5

In [15]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7662 entries, 0 to 7661
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  7662 non-null   object 
 1   Year             7662 non-null   int64  
 2   Value            7662 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 239.6+ KB


# Answer : 4 columns, 7662 rows; data types are country - object; year - int; value and value footnotes - float

# Q6

In [21]:
gdp_dfn = gdp_df.drop(columns=['Value Footnotes'])

In [23]:
gdp_dfn.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7662 entries, 0 to 7661
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  7662 non-null   object 
 1   Year             7662 non-null   int64  
 2   Value            7662 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 179.7+ KB


In [52]:
gdp_fna = gdp_dfn.rename(columns={'Country or Area': 'Country', 'Value': 'GDP_Per_Capita'})

# Q7

In [38]:
gdp_dfn['Country'].value_counts()

KeyError: 'Country'

In [39]:
print(nullq)

      Country or Area   Year  Value
0               False  False  False
1               False  False  False
2               False  False  False
3               False  False  False
4               False  False  False
...               ...    ...    ...
7657            False  False  False
7658            False  False  False
7659            False  False  False
7660            False  False  False
7661            False  False  False

[7662 rows x 3 columns]
